### ISSUE TITLE
Move filter options to new `form` object

### ISSUE THREAD

**Issue by jepoorbaugh** [jncraton/course-search#129]
The filter options should be moved to a new section of the app above the table. For accessibility reasons, this section should be defined by a form object that, when submitted, applies each filter to the table. In addition, changing the value of the inputs within the form should automatically apply the filters again, either through resubmitting the form or by calling the same method as when the form is submitted. 

Here is a diagram for how the table should look. 
![table_diagram](https://github.com/user-attachments/assets/87d22936-5850-49a7-9d73-64d04a95c207)
